src/backend/pipeline/cont_scheduler.c

```diff
@@ -703,12 +720,18 @@ ContQuerySchedulerMain(int argc, char *argv[])
 		 * Wait until naptime expires or we get some type of signal (all the
 		 * signal handlers will wake us by calling SetLatch).
 		 */
-		rc = WaitLatch(&MyProc->procLatch, WL_LATCH_SET | WL_POSTMASTER_DEATH, 0);
+		rc = WaitLatch(&MyProc->procLatch, WL_LATCH_SET | WL_POSTMASTER_DEATH | WL_TIMEOUT, 1000);
 
 		ResetLatch(&MyProc->procLatch);
 
 		DisableCatchupInterrupt();
 
+		if (rc & WL_TIMEOUT)
```

Comment:
This place is probably fine. What happens if this clean up doesn't happen? Other than redundant entries in `pipeline_query`?